### PR TITLE
Add g3log library

### PR DESCRIPTION
**Pull request by arnesor**
Specify library name and version:  **g3log/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Add new package g3log.

### CODE REVIEW

File: recipes/g3log/config.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+versions:
+  "1.3.2-86":
+    folder: all
+  "1.3.2":
+    folder: all
+  "20170108":
```
Comment (Croydon) [inline recipes/g3log/config.yml:6]: Please only add date versions when they project does no formal releases at all.

If you need a new version please ask first upstream for a new one.

Mixing date versions and actual versions numbers would be very confusing

File: recipes/g3log/config.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+versions:
+  "1.3.2-86":
```
Comment (Croydon) [inline recipes/g3log/config.yml:2]: What is the difference to `1.3.2`  and why do you want this?

File: recipes/g3log/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        os.rename("g3log-{}".format(dir_postfix), self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.compiler != "Visual Studio":
+            del self.options.enable_vectored_exception_handling
+            del self.options.debug_break_at_fatal_signal
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        self.bind_cmake_bool_option(cmake, "shared", "G3_SHARED_LIB")
```
Comment (uilianries) [inline recipes/g3log/all/conanfile.py:44]:
```suggestion
		cmake.definitions["G3_SHARED_LIB"] = self.options.share
```

same for all others

File: recipes/g3log/config.yml (hunk 1 above)
Comment (arnesor) [reply recipes/g3log/config.yml:6]: OK, I can remove the date version.

File: recipes/g3log/config.yml (hunk 2 above)
Comment (arnesor) [reply recipes/g3log/config.yml:2]: The g3log project has the last years been calculating their version number in CMakeLists.txt, based on some git properties, see [https://github.com/KjellKod/g3log/blob/master/CMakeLists.txt#L94](url), line 94-103. The last GitHub release, 1.3.2, is over two years old, and I don't know if they are doing those kind of releases anymore, or if they just stick to their current version scheme. At work we need a newer version than the two year old 1.3.2. There has been a lot of fixes and improvements and we use version 1.3.2-86 from Mars 6, 2020. That's why I added this version.

File: recipes/g3log/config.yml (hunk 1 above)
Comment (uilianries) [reply recipes/g3log/config.yml:6]: https://github.com/conan-io/conan-center-index/wiki/FAQ#what-version-should-packages-use-for-libraries-without-official-releases

File: recipes/g3log/config.yml (hunk 2 above)
Comment (Croydon) [reply recipes/g3log/config.yml:2]: https://github.com/KjellKod/g3log/issues/338

File: recipes/g3log/config.yml (hunk 1 above)
Comment (arnesor) [reply recipes/g3log/config.yml]: Fixed in 0585deb

File: recipes/g3log/all/conanfile.py (hunk 3 above)
Comment (arnesor) [reply recipes/g3log/all/conanfile.py]: Fixed in 0585deb

File: recipes/g3log/config.yml (hunk 1 above)
Comment (arnesor) [reply recipes/g3log/config.yml]: Sorry, wrong commit reference. Fixed in ffa6445

File: recipes/g3log/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    def build(self):
+        if "patches" in self.conan_data and self.version in self.conan_data["patches"]:
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        if self.settings.os == "Windows":
```
Comment (uilianries) [inline recipes/g3log/all/conanfile.py:88]: Did you try `cmake.install()` ?

Comment (arnesor) [reply recipes/g3log/all/conanfile.py:88]: Yes, that was my first approach. With version 1.3.2 I struggled quite a bit with link errors, wrong CMAKE_INSTALL_LIBDIR, tried to patch, still errors and then I went for the current fallback solution. The CMake part of g3log as improved quite a bit to version 1.3.2-86, and with that version `cmake.install()` works. But I get some hook errors: `[CMAKE-MODULES-CONFIG-FILES (KB-H016)]` and `[CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)]`.

Comment (uilianries) [reply recipes/g3log/all/conanfile.py:88]: Understood. These errors are regular, the regular approach is running cmake.install() and removing the folders which are not acceptable in CCI: https://github.com/conan-io/conan-center-index/blob/265958ce30706b359000b42dab360687175560a7/recipes/ogg/all/conanfile.py#L47

Comment (arnesor) [reply recipes/g3log/all/conanfile.py]: I gave it another try this evening, and this time I was able to fix it. I had to override CMAKE_INSTALL_LIBDIR with CACHE FORCE and change type from PATH to STRING. It now works locally for me with cmake.install() for both version 1.3.2 and 1.3.2-86. I have pushed the changes and hopefully it works for the other platforms as well.

Comment (arnesor) [reply recipes/g3log/all/conanfile.py]: I finally got it to work with `cmake.install()`. Fixed in e1e23cb.

File: recipes/g3log/config.yml (hunk 2 above)
Comment (gir489) [reply recipes/g3log/config.yml:2]: @Croydon The g3log project is officially in support mode. Meaning, pull requests will be considered and merged if required, but the author is not going to fix bugs himself anymore. The age of the release is irrelevant, given the recent commits are still. As @arnesor mentioned, the project is only doing support releases, and the age is indicative of that.  The commits in master are untested and potentially dangerous to have in a production environment. If you are bold enough, you can test them, but I highly suggest you don't.

**Pull request merged**